pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -285,11 +287,37 @@ func NewKubeGenericRuntimeManager(
 			os.Exit(1)
 		}
 	}
-	kubeRuntimeManager.keyring = credentialprovider.NewDefaultDockerKeyring()
 
+	var imageGCHooks []images.PostImageGCHook
+	var imagePullManager imagepullmanager.ImagePullManager = &imagepullmanager.NoopImagePullManager{}
+	if utilfeature.DefaultFeatureGate.Enabled(features.KubeletEnsureSecretPulledImages) {
+		imagePullCredentialsVerificationPolicy, err := imagepullmanager.NewImagePullCredentialVerificationPolicy(
+			kubeletconfiginternal.ImagePullCredentialsVerificationPolicy(imagePullsCredentialVerificationPolicy),
+			preloadedImagesCredentialVerificationWhitelist)
+
+		if err != nil {
+			return nil, nil, err
+		}
+
+		fsRecordAccessor, err := imagepullmanager.NewFSPullRecordsAccessor(rootDirectory)
```

Comment:
the PR for docs will be needed to update this: https://kubernetes.io/docs/reference/node/kubelet-files/

Reply:
the docs PR for the feature can be found here: https://github.com/kubernetes/website/pull/49886

Reply:
Added a note to that PR to update the kubelet-files